packages/opentelemetry-exporter-jaeger/src/jaeger.ts

```diff
@@ -31,21 +32,35 @@ export class JaegerExporter implements SpanExporter {
   private readonly _onShutdownFlushTimeout: number;
 
   constructor(config: jaegerTypes.ExporterConfig) {
-    this._logger = config.logger || new NoopLogger();
-    const tags: jaegerTypes.Tag[] = config.tags || [];
+    const localConfig = Object.assign({}, config)
+    this._logger = localConfig.logger || new NoopLogger();
+    const tags: jaegerTypes.Tag[] = localConfig.tags || [];
     this._onShutdownFlushTimeout =
-      typeof config.flushTimeout === 'number' ? config.flushTimeout : 2000;
+      typeof localConfig.flushTimeout === 'number' ? localConfig.flushTimeout : 2000;
 
-    config.host = config.host || process.env.JAEGER_AGENT_HOST;
+    // https://github.com/jaegertracing/jaeger-client-node#environment-variables
+    // By default, the client sends traces via UDP to the agent at localhost:6832. Use JAEGER_AGENT_HOST and
+    // JAEGER_AGENT_PORT to send UDP traces to a different host:port. If JAEGER_ENDPOINT is set, the client sends traces
+    // to the endpoint via HTTP, making the JAEGER_AGENT_HOST and JAEGER_AGENT_PORT unused. If JAEGER_ENDPOINT is secured,
+    // HTTP basic authentication can be performed by setting the JAEGER_USER and JAEGER_PASSWORD environment variables.
+    if (localConfig.endpoint) {
+      localConfig.endpoint = localConfig.endpoint || process.env.JAEGER_ENDPOINT;
```

Comment:
this will never use the env

Reply:
This should fix it: 8d518008f724071807a26ab3ce6cfe0a7ce2cf4f